src/Plugin/Block/FooterBlock.php

```diff
@@ -81,9 +93,59 @@ public function build() {
 
     NestedArray::setValue($build, ['#corporate_footer', 'bottom_links'], $config->get('bottom_links'));
 
+    $general_links = $this->retriveCustomFooterLinks('footer_general_link', $cache);
+    $social_links = $this->retriveCustomFooterLinks('footer_social_link', $cache);
+    $site_info_config = $this->configFactory->get('system.site');
+    $cache->addCacheableDependency($site_info_config);
+    $site_identity = $site_info_config->get('name');
+
+    if (!empty($site_identity) && !empty($social_links) && !empty($general_links)) {
+      NestedArray::setValue($build, ['#custom_footer', 'site_identity'], $site_identity);
+      NestedArray::setValue($build, ['#custom_footer', 'social_links'], $social_links);
+      NestedArray::setValue($build, ['#custom_footer', 'other_links'], $general_links);
+    }
+
     $cache->applyTo($build);
 
     return $build;
   }
 
+  /**
+   * Retrieve a list of custom footer link configuration entities.
+   *
+   * @param string $type
+   *   Type of configs.
+   * @param \Drupal\Core\Cache\CacheableMetadata $cache
+   *   The Cacheable metadata abject.
+   *
+   * @return array
+   *   The array of links.
+   */
+  protected function retriveCustomFooterLinks(string $type, CacheableMetadata &$cache): array {
+    $links = [];
+    /** @var \Drupal\Core\Entity\EntityStorageInterface $links_storage */
+    $links_storage = $this->entityTypeManager->getStorage($type);
+    $cache->addCacheTags($links_storage->getEntityType()->getListCacheTags());
+    if ($links_storage->getQuery()->count()->execute() > 0) {
```

Comment:
Don't have to run two queries here. Just query them by weight and just return if there are no results. Then you load if you have IDs.